git diff
diff --git a/CMakeLists.txt b/CMakeLists.txt
index b5bfaf6..12ddfda 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -122,6 +122,10 @@ IF(CMAKE_SYSTEM MATCHES "Darwin")
         _append(ACNG_COMPFLAGS -D_DARWIN_C_SOURCE)
         #it seems like clang++ still needs to be explicitly told to support c++11
         _append(ACNG_CXXFLAGS -std=c++11)
+        # look for fmemopen(), 'REQUIRED' option will cause cmake fail if not present
+        pkg_check_modules(FMEMOPEN REQUIRED libfmemopen)
+        list(APPEND CompLibs fmemopen)
+        _append(EXTRA_LIBS_ACNGFS fmemopen)
 ENDIF()

 IF(CMAKE_SYSTEM MATCHES "CYGWIN")
diff --git a/fs/CMakeLists.txt b/fs/CMakeLists.txt
index 49791e3..08d23b8 100644
--- a/fs/CMakeLists.txt
+++ b/fs/CMakeLists.txt
@@ -12,6 +12,10 @@ FILE(READ ../test/build/HAVE_FUSE_25.cc TESTSRC)
 SET(CMAKE_REQUIRED_FLAGS "${ACNG_COMPFLAGS} ${acngfs_cflags}")
 CHECK_CXX_SOURCE_COMPILES("${TESTSRC}" HAVE_FUSE_25)

+IF(CMAKE_SYSTEM MATCHES "Darwin")
+        _append(EXTRA_LIBS_ACNGFS fmemopen)
+ENDIF()
+
 if(fuse_FOUND AND HAVE_FUSE_25)
         # not linking acngstuff lib, too many dependencies, so building a custom variant with MINIBUILD flag
         list(APPEND fsSRCS httpfs.cc ../source/lockable.cc ../source/header.cc ../source/caddrinfo.cc ../source/acbuf.cc ../source/acfg.cc ../source/acfg_defaults.cc ../source/tcpconnect.cc ../source/dlcon.cc ../source/fileitem.cc ../source/meta.cc ../source/fileio.cc)
diff --git a/source/meta.cc b/source/meta.cc
index 30f74c7..6f631eb 100644
--- a/source/meta.cc
+++ b/source/meta.cc
@@ -9,6 +9,7 @@
 #include <cstdio>
 #include <iostream>
 #include <algorithm>
+#include <fmemopen.h>

 #ifdef HAVE_WORDEXP
 #include <wordexp.h>
