diff --git a/CMakeLists.txt b/CMakeLists.txt
index b5bfaf6..78cefbb 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -122,6 +122,9 @@ IF(CMAKE_SYSTEM MATCHES "Darwin")
         _append(ACNG_COMPFLAGS -D_DARWIN_C_SOURCE)
         #it seems like clang++ still needs to be explicitly told to support c++11
         _append(ACNG_CXXFLAGS -std=c++11)
+        # look for fmemopen(), 'REQUIRED' option will cause cmake fail if not present
+        pkg_check_modules(FMEMOPEN REQUIRED libfmemopen)
+        list(APPEND CompLibs fmemopen)
 ENDIF()

 IF(CMAKE_SYSTEM MATCHES "CYGWIN")
diff --git a/fs/CMakeLists.txt b/fs/CMakeLists.txt
index 49791e3..36b66c5 100644
--- a/fs/CMakeLists.txt
+++ b/fs/CMakeLists.txt
@@ -26,10 +26,11 @@ if(fuse_FOUND AND HAVE_FUSE_25)
         if(HAVE_DLOPEN)
                 _append(EXTRA_LIBS_ACNGFS dl)
         endif()
+        IF(CMAKE_SYSTEM MATCHES "Darwin")
+                _append(EXTRA_LIBS_ACNGFS fmemopen)
+        ENDIF()
         TARGET_LINK_LIBRARIES(acngfs ${fuse_LDFLAGS} ${BaseNetworkLibs} ${SSL_LIB_LIST} ${EXTRA_LIBS_ACNGFS} ${CMAKE_THREAD_LIBS_INIT})

 else(fuse_FOUND AND HAVE_FUSE_25)
         message("- FUSE not found or not compatible, not building acngfs")
 endif(fuse_FOUND AND HAVE_FUSE_25)
-
-
diff --git a/source/meta.cc b/source/meta.cc
index 30f74c7..6f631eb 100644
--- a/source/meta.cc
+++ b/source/meta.cc
@@ -9,6 +9,7 @@
 #include <cstdio>
 #include <iostream>
 #include <algorithm>
+#include <fmemopen.h>

 #ifdef HAVE_WORDEXP
 #include <wordexp.h>
